storage/src/test/java/org/apache/kafka/tiered/storage/integration/LocalLogStartOffsetTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.tiered.storage.integration;
+
+import org.apache.kafka.tiered.storage.TieredStorageTestBuilder;
+import org.apache.kafka.tiered.storage.TieredStorageTestHarness;
+import org.apache.kafka.tiered.storage.specs.KeyValueSpec;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.kafka.common.utils.Utils.mkEntry;
+import static org.apache.kafka.common.utils.Utils.mkMap;
+
+/**
+ * This test exercises the consumption of remote data on a leader that was previously a follower
+ * and had rebuilt its auxiliary state from remote storage.
+ */
+public final class LocalLogStartOffsetTest extends TieredStorageTestHarness {
```

Comment:
This is one more variant of [ReassignReplicaExpandTest](https://github.com/apache/kafka/blob/trunk/storage/src/test/java/org/apache/kafka/tiered/storage/integration/ReassignReplicaExpandTest.java) where we don't produce message post the leader switch. Can we rename the test to `ReassignReplicaExpandTest1` (or) something similar?

Reply:
Can we rename it as `ReassignToEmptyReplica` or some other better name?

Reply:
Renamed to `ReassignToEmptyReplicaTest`